dotnet/Miscreant/AesCtr.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Security.Cryptography;
+
+namespace Miscreant
+{
+	/// <summary>
+	/// Counter (CTR) mode, defined in NIST Special Publication
+	/// <see href="https://csrc.nist.gov/publications/detail/sp/800-38a/final">SP 800-38A</see>.
+	/// </summary>
+	public sealed class AesCtr : IDisposable
```

Comment:
Should this (and potentially `AesSiv` and `AesPmacSiv`) implement `ICryptoTransform`?

https://msdn.microsoft.com/en-us/library/system.security.cryptography.icryptotransform(v=vs.110).aspx

Reply:
`AesSiv` and `AesPmacSiv` don't really fit into that interface. `AesCtr` could implement the interface, but the implementation is much more simple this way. I was thinking about this, too - `AesCtr` (and `AesPmac` and `AesCmac` also) should not actually be exposed to the public, but given that such important primitive is not available in the standard library, maybe it would be useful thing to have it as a part of Miscreant?

Reply:
I've gravitated towards exposing CMAC and PMAC in the other implementations. They are high-quality PRFs and there's nothing scary about exposing them.

AES-CTR is a bit more questionable.

Reply:
I made AES-CTR internal in the latest commit.